Cleanup InternalsVisibleTo attributes

olegsych opened the pull request:
- Extract public and test keys into reusable constants
  - Create a new project, `Microsoft.ServiceFabric.Constants` to hold compile-time constants that should produce runtime-dependencies.
  - Move the Signature class from the `TestFramework` to the `Constants` project and rename it to `AssemblyInfo`
  - Replace hard-coded public keys in all projects with the `AssemblyInfo` constants.
- Remove unnecessary `InternalsVisibleTo` attributes
  - Assemblies produced by projects in this repository don't use the `TestKey`
  - Assemblies that no longer exist
- Extract the `DynamicProxyGenAssembly2` literal from all projects to `Constants.AssemblyInfo`.
- Rename `Friend*.cs` files to `AssemblyInfo.cs`
  Old file names used inconsistent capitalization, some used underscores. `AssemblyInfo.cs` is a widely-used convention established by the Visual Studio project templates.

olegsych committed:
Extract public and test keys into reusable constants

- Create a new project, Microsoft.ServiceFabric.Constants to hold compile-time constants that should produce runtime-dependencies.
- Move the Signature class from the TestFramework to the Constants project and rename it to AssemblyInfo, following the widely used convention established by the Visual Studio project templates.
- Replace hard-coded public keys in all projects with the AssemblyInfo constants.

olegsych committed:
Remove unnecessary InternalsVisibleTo attributes

- Assemblies produced by projects in this repository don't use the TestKeys
- Assemblies that no longer exist

olegsych committed:
Rename Friend*.cs files to AssemblyInfo.cs

Our Friend.cs file names used inconsistent capitalization, some used underscores. AssemblyInfo is a widely-used convention established by the Visual Studio project templates.

ilijagrbic commented on src/Microsoft.ServiceFabric.Diagnostics/AssemblyInfo.cs:
```diff
@@ -4,44 +4,37 @@
-// The key strings are taken from snk files referenced in $(SrcRoot)\prod\StrongNameKeys.props
-// WindowsPublicKey = 0024000004800000940000000602000000240000525341310004000001000100b5fc90e7027f67871e773a8fde8938c81dd402ba65b9201d60593e96c492651e889cc13f1415ebb53fac1131ae0bd333c5ee6021672d9718ea31a8aebd0da0072f25d87dba6fc90ffd598ed4da35e44c398c454307e8e33b8426143daec9f596836f97c8f74750e5975c64e2189f45def46b2a2b1247adc3652bf5c308055da9
-// InsecurePublicKey = 00240000048000009400000006020000002400005253413100040000010001003f8c902c8fe7ac83af7401b14c1bd103973b26dfafb2b77eda478a2539b979b56ce47f36336741b4ec52bbc51fecd51ba23810cec47070f3e29a2261a2d1d08e4b2b4b457beaa91460055f78cc89f21cd028377af0cc5e6c04699b6856a1e49d5fad3ef16d3c3d6010f40df0a7d6cc2ee11744b5cfb42e0f19a52b8a29dc31b0
+[assembly: InternalsVisibleTo("ServiceFabricTracingStandaloneTest" + TestKey)]
+[assembly: InternalsVisibleTo("ServiceFabricTracingTest" + TestKey)]
+[assembly: InternalsVisibleTo("Microsoft.ServiceFabric.Actors" + PublicKey)]
+[assembly: InternalsVisibleTo("Microsoft.ServiceFabric.Services" + PublicKey)]
+[assembly: InternalsVisibleTo("CoordinatorService" + PublicKey)]
+[assembly: InternalsVisibleTo("NodeAgentSFUtilityDll" + PublicKey)]
+[assembly: InternalsVisibleTo("NodeAgentSFUtility" + PublicKey)]
+[assembly: InternalsVisibleTo("NodeAgentWorkerLib" + PublicKey)]
+[assembly: InternalsVisibleTo("NodeAgentService" + PublicKey)]
+[assembly: InternalsVisibleTo("CoordinatorServiceTest" + TestKey)]
+[assembly: InternalsVisibleTo("NodeAgentSFUtilityTest" + TestKey)]
+[assembly: InternalsVisibleTo("TelemetryLibTest" + TestKey)]
+[assembly: InternalsVisibleTo("EventsValidationTest" + PublicKey)]
+[assembly: InternalsVisibleTo("AzureFilesVolumePlugin" + PublicKey)]
+[assembly: InternalsVisibleTo("AzureFilesVolumePluginSetup" + PublicKey)]
+[assembly: InternalsVisibleTo("SFVolumeDiskDebugAgentCommon" + PublicKey)]
+[assembly: InternalsVisibleTo("NodeAgentServiceSetup" + PublicKey)]
+[assembly: InternalsVisibleTo("Microsoft.ServiceFabric.Diagnostics.Tests" + PublicKey)]
+[assembly: InternalsVisibleTo("Microsoft.ServiceFabric.Actors.Tests" + PublicKey)]
+[assembly: InternalsVisibleTo("Microsoft.ServiceFabric.Services.Tests" + PublicKey)]
+[assembly: InternalsVisibleTo("Microsoft.ServiceFabric.TestFramework" + PublicKey)]
+[assembly: InternalsVisibleTo(DynamicProxyGenAssembly2)]
```
Why are we not creating `AssemblyInfo.cs` in this project instead of using old `_friend.cs`? I see that for most project we are using `AssemblyInfo.cs`, except for `Diagnostics` and `Services.Wcf`.

olegsych replied:
Good idea. I already started on this in #480

olegsych replied:
Actually, it's a part of this PR. Updated.

ilijagrbic commented on src/Constants/Microsoft.ServiceFabric.Constants.csproj:
```diff
@@ -0,0 +1,10 @@
+    The assembly produced by this project should not be included in any of the NuGet packages.
+  -->
+  <PropertyGroup>
+    <TargetFrameworks>netstandard2.0</TargetFrameworks>
```
Why are we using netstandard here instead of having net462 and net8.0 (as in other projects)? I suspect there is a good reason, I just haven't figured it out. :)

olegsych replied:
Since this project will have only constants, making it netstandard2.0 allows us to build it only once. Not much of an optimization, but still.

ilijagrbic reviewed (changes requested):
Hey @olegsych , awesome changes, thanks for doing this. I think the new way to add `InternalsVisibleTo` is much more readable and scalable. 
Before merging I would just consider renaming `_friend.cs` files from Diagnostics/Services.Wcf project into `AsseblyInfo.cs` for consistency through all projects, unless there is some other reason for this that I'm missing

plave0 commented on src/Microsoft.ServiceFabric.Diagnostics/Microsoft.ServiceFabric.Diagnostics.csproj:
```diff
@@ -7,4 +7,7 @@
     <PackageReference Include="Microsoft.ServiceFabric" />
   </ItemGroup>
+  <ItemGroup>
+    <ProjectReference Include="$(RepoRoot)src\Constants\Microsoft.ServiceFabric.Constants.csproj" />
```
nit: We should use absolute or relative paths for project references consistently throughout the repo. We use relative paths in other projects, but in this project you used an absolute path.

olegsych replied:
Good point. It looks like we're using relative paths in the src projects and RepoRoot in the test projects.

olegsych committed:
Extract DynamicProxyGenAssembly2 from projects to Constants.AssemblyInfo

olegsych commented:
Thanks for the review @plave0 and @ilijagrbic. I've addressed your comments. Could you re-approve? I think @ilijagrbic's approval is required because he requested changes.
